portfolio/src/main/webapp/script.js

```diff
@@ -91,4 +91,103 @@ function deleteDataAndFetch() {
   let commentsEl = document.getElementById("all-comments");
   commentsEl.innerHTML = "";
   fetch('/delete-data', {method: "POST"}).then(() => fetchFromData());
+  window.location.href = '/index.html';
+}
+
+/** Creates a map and adds it to the page. */
+function createMap() {
+  const OFFICE_LOC = {lat: 32.0700, lng: 34.7941};
+  const OFFICE_DESCRIPTION = 'Tel Aviv Google office';
+
+  // The map
+  const map = new google.maps.Map(
+    document.getElementById('map'), {
+      center: OFFICE_LOC,
+      zoom: 15,
+      mapTypeControlOptions: { mapTypeIds: ['roadmap', 'satellite', 'hybrid', 'terrain']}
+    }
+  );
+
+  // A marker
+  const officeMarker = new google.maps.Marker({
+    position: OFFICE_LOC,
+    map: map,
+    title: OFFICE_DESCRIPTION,
+    animation: google.maps.Animation.DROP
+  });
+
+  // The marker's info window
+  const infoWindow = new google.maps.InfoWindow({
+    content: OFFICE_DESCRIPTION
+  });
+
+  // when marker is clicked, the info window is opened
+  officeMarker.addListener("click", () => {
+    infoWindow.open(map, officeMarker);
+  });
+    
+  // When center of the map changes, after 3 seconds we pan back to the marker.
+  map.addListener('center_changed', function() {
+    window.setTimeout(function() {
+      map.panTo(officeMarker.getPosition());
+    }, 3000);
+  });
+}
+
+function createMarketMap() {
+    fetch('/farmer-market').then(response => response.json()).then((markets) => {
+    // Create map
+    const map = new google.maps.Map(
+      document.getElementById('map'),
+      {center: {lat: 39.842507, lng: -97.058318}, zoom: 2}
+    );
+
+    // Add markers to the map
+    markets.forEach((market) => {
+      const marker = new google.maps.Marker(
+        {position: {lat: market.lat, lng: market.lng}, map: map}
+        );
+
+      const marketContent = "Market's name: " + market.name + ". Website: " + market.website;
+      const infowindow = new google.maps.InfoWindow({
+        content: marketContent
+      });
+
+      marker.addListener("click", () => {
+        infowindow.open(map, marker); 
+      });
+    });
+  });
+}
+
+// Loading the google chart preperation
+google.charts.load('current', {'packages':['corechart']});
+google.charts.setOnLoadCallback(drawChart);
+
+/** Creates a chart and adds it to the page. */
+function drawChart() {  
```

Comment:
I believe this function should be called when comments are added/removed from your page. 
This way when a user adds/removes a comment, the chart is updated accordingly. 
In later stages, you can do it as a single HTTP request (user adds comment -> HTTP response contains chart data -> redraw chart).
What do you think?

Reply:
SG, done (not yet in a single request).